src/classes/RD2_ElevateIntegrationService.cls

```diff
@@ -334,7 +370,200 @@ public inherited sharing class RD2_ElevateIntegrationService {
+        * @param response UTIL_Http.Response
+        * @return Boolean
+        */
+        private Boolean isQualifiedForRetry(
+            RD2_ElevateIntegrationService.CommitmentRD commitmentRD,
+            UTIL_Http.Response response
+        ) {
+            return hasRetriesRemaining(commitmentRD) && response.statusCode != UTIL_Http.STATUS_CODE_NOT_FOUND;
+        }
+
+        /**
+        * @description Returns true if commitmentRD has retries remaining.
+        * @param commitmentRD RD2_ElevateIntegrationService.CommitmentRD
+        * @return Boolean
+        */
+        private Boolean hasRetriesRemaining(RD2_ElevateIntegrationService.CommitmentRD commitmentRD) {
+            return commitmentRD.retriesRemaining > 0;
+        }
+
+        /**
+        * @description Handles error response.
+        * @param commitmentRD RD2_ElevateIntegrationService.CommitmentRD
+        * @param response UTIL_Http.Response
+        */
+        private void handleError(
+            RD2_ElevateIntegrationService.CommitmentRD commitmentRD,
+            UTIL_Http.Response response
+        ) {
+            commitmentRD.isConfirmationRequired = isConfirmationNeeded(response);
+            queuedCommitmentRDs.add(commitmentRD);
+        }
+
+        /**
+        * @description Returns true if response indicates that Elevate status confirmation is needed.
+        * @param response UTIL_Http.Response
+        * @return Boolean
+        */
+        private Boolean isConfirmationNeeded(UTIL_Http.Response response) {
+            return response.statusCode == UTIL_Http.STATUS_CODE_REQUEST_TIMEOUT
+                || response.statusCode == UTIL_Http.STATUS_CODE_GATEWAY_TIMEOUT
+                || response.statusCode == UTIL_Http.STATUS_CODE_INTERNAL_SERVER_ERROR
+                || response.statusCode == UTIL_Http.STATUS_CODE_SERVICE_UNAVAILABLE
+                || response.statusCode == UTIL_Http.STATUS_CODE_CONFLICT;
+        }
+
+        private List<RD2_ElevateIntegrationService.CommitmentRD> prepCommitmentsForRetry() {
+            List<RD2_ElevateIntegrationService.CommitmentRD> retryCommitments = new List<RD2_ElevateIntegrationService.CommitmentRD>();
+
+            if (!queuedCommitmentRDs.isEmpty()) {
+                for (RD2_ElevateIntegrationService.CommitmentRD retryCommitment : queuedCommitmentRDs) {
+                    if (retryCommitment.retriesRemaining > 0) {
+                        retryCommitment.retriesRemaining--;
+                        retryCommitments.add(retryCommitment);
+                    }
+                }
+            }
+            return retryCommitments;
+        }
+
+        /**
+        * @description Handles error logging and setting final Status Reason
+        * @param handledCommitmentRDs List of RD2_ElevateIntegrationService.CommitmentRD
+        * @param handledResponses List of UTIL_Http.Response
+        */
+        private void logErrors(
```

Comment:
`logErrors()` id setting the "Status Reason" for records that succeeded, are canceled already or they failed in addition to logging errors.